lib/heuristics/concerns/scheduling_data_initialisation.rb

```diff
@@ -152,32 +152,30 @@ def adapt_services_data(vrp)
 
     @to_plan_service_ids = []
     vrp.points.each{ |point|
-      same_located_set = vrp.services.select{ |service|
-        (service.activity ? [service.activity] : service.activities).any?{ |activity|
-          activity.point.id == point.id
-        }
-      }.sort_by(&:visits_number)
+      same_located_set = @services_data.select{ |_id, data| data[:points_ids].include?(point.id) }.sort_by{ |_id, data| data[:raw].visits_number }
 
       next if same_located_set.empty?
 
-      raise OptimizerWrapper.UnsupportedProblemError, 'Same_point_day is not supported if a set has one service with several activities' if same_located_set.any?{ |s| !s.activities.empty? }
+      raise OptimizerWrapper.UnsupportedProblemError, 'Same_point_day is not supported if a set has one service with several activities' if same_located_set.any?{ |id, data| data[:points_ids].size > 1 }
+
+      raise OptimizerWrapper::UnsupportedProblemError, 'Same_point_day is not supported if frequences of a set have no common multiplexer' unless common_divisor?(same_located_set.collect{ |_id, data| data[:heuristic_period] })
```

Comment:
Wouldn't it be better to move these checks inside the `self.check_consistency(hash)` function. Because otherwise we accept the VRP and return `200` but then raise an error.